crates/pallets/starknet/src/lib.rs

```diff
@@ -1079,31 +1081,27 @@ impl<T: Config> Pallet<T> {
     pub fn estimate_fee(transactions: Vec<UserTransaction>) -> Result<Vec<(u64, u64)>, DispatchError> {
         let chain_id = Self::chain_id();
 
-        let execution_results = execute_txs_and_rollback::<T>(
+        let execution_infos = execute_txs_until_first_failed_and_rollback::<T>(
             &transactions,
             &Self::get_block_context(),
             chain_id,
             &mut RuntimeExecutionConfigBuilder::new::<T>().with_query_mode().build(),
-        )?;
-
-        let mut results = vec![];
-        for res in execution_results {
-            match res {
-                Ok(tx_exec_info) => {
-                    log!(info, "Successfully estimated fee: {:?}", tx_exec_info);
-                    if let Some(l1_gas_usage) = tx_exec_info.actual_resources.0.get("l1_gas_usage") {
-                        results.push((tx_exec_info.actual_fee.0 as u64, *l1_gas_usage));
-                    } else {
-                        return Err(Error::<T>::TransactionExecutionFailed.into());
-                    }
-                }
-                Err(e) => {
-                    log!(info, "Failed to estimate fee: {:?}", e);
-                    return Err(Error::<T>::TransactionExecutionFailed.into());
-                }
-            }
-        }
-        Ok(results)
+        )?
+        .map_err(|_| DispatchError::from(Error::<T>::TransactionExecutionFailed))?;
```

Comment:
can we log the error

Reply:
we would be logging `PlaceHolderErrorTypeForFailedStarknetExecution`. Nothing really intresting